libpromises/evalfunction.c

```diff
@@ -8668,7 +8668,11 @@ static FnCallResult FnCallIsReadable(ARG_UNUSED EvalContext *const ctx,
                 "Read operation timed out, exceeded %ld seconds.", path,
                 timeout);
 
+#ifdef HAVE_PTHREAD_CANCEL
             ret = pthread_cancel(thread_data.thread);
+#else
+            ret = pthread_kill(thread_data.thread, 0);
```

Comment:
Is it safe to join a "killed" thread? Will this call the signal handler installed by the agent. If so, is this fine? Also the man pthread_kill(3) says: 

> If sig is 0, then no signal is sent, but error checking is still performed.

Reply:
I should read your blog post, man pages and then try out `isreadable()` and see how it goes on termux. :)

Reply:
Will read https://cfengine.com/blog/2023/check-if-file-is-readable/ and then proceed.

Reply:
Looks like there is already a patch in termux-packages we may be able to pull in.

https://github.com/termux/termux-packages/blob/master/packages/cfengine/pthread_cancel.patch